### ISSUE TITLE
[Bug] I cannot complete the installation of Hestia on Ubuntu 22.04

### ISSUE THREAD

**Issue by leandroboari** [hestiacp/hestiacp#3082]
### Describe the bug

I cannot complete the installation of Hestia on Ubuntu 22.04.

### Tell us how to replicate the bug

Step 1: Make sure that your server is running a supported operating system (amd64 or arm64):

Ubuntu 22.04 LTS running on Vultr

Step 2: Log in as the system user (root):

root

Step 3: Download the installer:

wget https://raw.githubusercontent.com/hestiacp/hestiacp/release/install/hst-install.sh

Step 4: Run the installer and follow the prompts:

bash hst-install.sh

### Which components are affected by this bug?

Control Panel Installation or Upgrade

### Hestia Control Panel Version

1.6.11

### Operating system

Ubuntu 22.04 LTS

### Log capture

```shell
Adding required repositories to proceed with installation:

[ * ] NGINX
[ * ] PHP
[ * ] Apache2
[ * ] MariaDB
[ * ] Hestia Control Panel

Updating currently installed packages, please wait.../
The installer is now downloading and installing all required packages.
NOTE: This process may take 10 to 15 minutes to complete, please wait...
E: Failed to fetch https://ppa.launchpadcontent.net/ondrej/php/ubuntu/pool/main/p/php8.0/php8.0-cli_8.0.25-1%2bubuntu22.04.1%2bdeb.sury.org%2b1_amd64.deb  Cannot initiate the connection to ppa.launchpadcontent.net:443 (2620:2d:4000:1::3e). - connect (101: Network is unreachable) Could not connect to ppa.launchpadcontent.net:443 (185.125.190.52), connection timed out [IP: 185.125.190.52 443]
E: Failed to fetch https://ppa.launchpadcontent.net/ondrej/php/ubuntu/pool/main/p/php8.0/php8.0-fpm_8.0.25-1%2bubuntu22.04.1%2bdeb.sury.org%2b1_amd64.deb  Cannot initiate the connection to ppa.launchpadcontent.net:443 (2620:2d:4000:1::3e). - connect (101: Network is unreachable) [IP: 185.125.190.52 443]
E: Failed to fetch https://ppa.launchpadcontent.net/ondrej/php/ubuntu/pool/main/p/php-apcu/php8.0-apcu_5.1.21%2b4.0.11-8%2bubuntu22.04.1%2bdeb.sury.org%2b1_amd64.deb  Cannot initiate the connection to ppa.launchpadcontent.net:443 (2620:2d:4000:1::3e). - connect (101: Network is unreachable) [IP: 185.125.190.52 443]
E: Failed to fetch https://ppa.launchpadcontent.net/ondrej/php/ubuntu/pool/main/p/php-imagick/php8.0-imagick_3.7.0-2%2bubuntu22.04.1%2bdeb.sury.org%2b2_amd64.deb  Cannot initiate the connection to ppa.launchpadcontent.net:443 (2620:2d:4000:1::3e). - connect (101: Network is unreachable) [IP: 185.125.190.52 443]
E: Failed to fetch https://ppa.launchpadcontent.net/ondrej/php/ubuntu/pool/main/p/php8.0/php8.0_8.0.25-1%2bubuntu22.04.1%2bdeb.sury.org%2b1_all.deb  Cannot initiate the connection to ppa.launchpadcontent.net:443 (2620:2d:4000:1::3e). - connect (101: Network is unreachable) [IP: 185.125.190.52 443]
E: Failed to fetch https://ppa.launchpadcontent.net/ondrej/php/ubuntu/pool/main/p/php8.0/php8.0-bcmath_8.0.25-1%2bubuntu22.04.1%2bdeb.sury.org%2b1_amd64.deb  Cannot initiate the connection to ppa.launchpadcontent.net:443 (2620:2d:4000:1::3e). - connect (101: Network is unreachable) [IP: 185.125.190.52 443]
E: Failed to fetch https://ppa.launchpadcontent.net/ondrej/php/ubuntu/pool/main/p/php8.0/php8.0-bz2_8.0.25-1%2bubuntu22.04.1%2bdeb.sury.org%2b1_amd64.deb  Cannot initiate the connection to ppa.launchpadcontent.net:443 (2620:2d:4000:1::3e). - connect (101: Network is unreachable) [IP: 185.125.190.52 443]
E: Failed to fetch https://ppa.launchpadcontent.net/ondrej/php/ubuntu/pool/main/p/php8.0/php8.0-curl_8.0.25-1%2bubuntu22.04.1%2bdeb.sury.org%2b1_amd64.deb  Cannot initiate the connection to ppa.launchpadcontent.net:443 (2620:2d:4000:1::3e). - connect (101: Network is unreachable) [IP: 185.125.190.52 443]
E: Failed to fetch https://ppa.launchpadcontent.net/ondrej/php/ubuntu/pool/main/p/php8.0/php8.0-gd_8.0.25-1%2bubuntu22.04.1%2bdeb.sury.org%2b1_amd64.deb  Cannot initiate the connection to ppa.launchpadcontent.net:443 (2620:2d:4000:1::3e). - connect (101: Network is unreachable) [IP: 185.125.190.52 443]
E: Failed to fetch https://ppa.launchpadcontent.net/ondrej/php/ubuntu/pool/main/p/php8.0/php8.0-imap_8.0.25-1%2bubuntu22.04.1%2bdeb.sury.org%2b1_amd64.deb  Cannot initiate the connection to ppa.launchpadcontent.net:443 (2620:2d:4000:1::3e). - connect (101: Network is unreachable) [IP: 185.125.190.52 443]
E: Failed to fetch https://ppa.launchpadcontent.net/ondrej/php/ubuntu/pool/main/p/php8.0/php8.0-intl_8.0.25-1%2bubuntu22.04.1%2bdeb.sury.org%2b1_amd64.deb  Cannot initiate the connection to ppa.launchpadcontent.net:443 (2620:2d:4000:1::3e). - connect (101: Network is unreachable) [IP: 185.125.190.52 443]
E: Failed to fetch https://ppa.launchpadcontent.net/ondrej/php/ubuntu/pool/main/p/php8.0/php8.0-ldap_8.0.25-1%2bubuntu22.04.1%2bdeb.sury.org%2b1_amd64.deb  Cannot initiate the connection to ppa.launchpadcontent.net:443 (2620:2d:4000:1::3e). - connect (101: Network is unreachable) [IP: 185.125.190.52 443]
E: Failed to fetch https://ppa.launchpadcontent.net/ondrej/php/ubuntu/pool/main/p/php8.0/php8.0-mbstring_8.0.25-1%2bubuntu22.04.1%2bdeb.sury.org%2b1_amd64.deb  Cannot initiate the connection to ppa.launchpadcontent.net:443 (2620:2d:4000:1::3e). - connect (101: Network is unreachable) [IP: 185.125.190.52 443]
E: Failed to fetch https://ppa.launchpadcontent.net/ondrej/php/ubuntu/pool/main/p/php8.0/php8.0-mysql_8.0.25-1%2bubuntu22.04.1%2bdeb.sury.org%2b1_amd64.deb  Cannot initiate the connection to ppa.launchpadcontent.net:443 (2620:2d:4000:1::3e). - connect (101: Network is unreachable) [IP: 185.125.190.52 443]
E: Failed to fetch https://ppa.launchpadcontent.net/ondrej/php/ubuntu/pool/main/p/php8.0/php8.0-pspell_8.0.25-1%2bubuntu22.04.1%2bdeb.sury.org%2b1_amd64.deb  Cannot initiate the connection to ppa.launchpadcontent.net:443 (2620:2d:4000:1::3e). - connect (101: Network is unreachable) [IP: 185.125.190.52 443]
E: Failed to fetch https://ppa.launchpadcontent.net/ondrej/php/ubuntu/pool/main/p/php8.0/php8.0-soap_8.0.25-1%2bubuntu22.04.1%2bdeb.sury.org%2b1_amd64.deb  Cannot initiate the connection to ppa.launchpadcontent.net:443 (2620:2d:4000:1::3e). - connect (101: Network is unreachable) [IP: 185.125.190.52 443]
E: Failed to fetch https://ppa.launchpadcontent.net/ondrej/php/ubuntu/pool/main/p/php8.0/php8.0-xml_8.0.25-1%2bubuntu22.04.1%2bdeb.sury.org%2b1_amd64.deb  Cannot initiate the connection to ppa.launchpadcontent.net:443 (2620:2d:4000:1::3e). - connect (101: Network is unreachable) [IP: 185.125.190.52 443]
E: Failed to fetch https://ppa.launchpadcontent.net/ondrej/php/ubuntu/pool/main/p/php8.0/php8.0-zip_8.0.25-1%2bubuntu22.04.1%2bdeb.sury.org%2b1_amd64.deb  Cannot initiate the connection to ppa.launchpadcontent.net:443 (2620:2d:4000:1::3e). - connect (101: Network is unreachable) [IP: 185.125.190.52 443]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?


========================================================================

[ * ] Configuring system settings...
[ * ] Configuring Hestia Control Panel...
cp: cannot stat '/usr/local/hestia/install/deb/sudo/admin': No such file or directory
chmod: cannot access '/etc/sudoers.d/admin': No such file or directory
cp: cannot stat '/usr/local/hestia/install/deb/logrotate/hestia': No such file or directory
ln: failed to create symbolic link '/usr/local/hestia/log': No such file or directory
touch: cannot touch '/usr/local/hestia/log/system.log': No such file or directory
touch: cannot touch '/usr/local/hestia/log/nginx-error.log': No such file or directory
touch: cannot touch '/usr/local/hestia/log/auth.log': No such file or directory
touch: cannot touch '/usr/local/hestia/log/backup.log': No such file or directory
chmod: cannot access '/usr/local/hestia/log': No such file or directory
chmod: cannot access '/var/log/hestia/*': No such file or directory
cp: cannot stat '/usr/local/hestia/install/deb/packages': No such file or directory
sed: can't read /usr/local/hestia/data/packages/*.pkg: No such file or directory
cp: cannot stat '/usr/local/hestia/install/deb/templates': No such file or directory
cp: cannot stat '/usr/local/hestia/install/deb/templates/web/unassigned/index.html': No such file or directory
cp: cannot stat '/usr/local/hestia/install/deb/templates/web/skel/document_errors/*': No such file or directory
cp: cannot stat '/usr/local/hestia/install/deb/firewall': No such file or directory
cp: cannot stat '/usr/local/hestia/install/deb/api': No such file or directory
[ * ] Generating default self-signed SSL certificate...
hst-install-ubuntu.sh: line 1301: /usr/local/hestia/bin/v-generate-ssl-cert: No such file or directory
[ * ] Adding SSL certificate to Hestia Control Panel...
sed: -e expression #1, char 3: unexpected `,'
sed: -e expression #1, char 1: unknown command: `,'
cp: cannot stat '/usr/local/hestia/install/deb/ssl/dhparam.pem': No such file or directory
[ * ] Enable SFTP jail...
Error: can't enable sftp jail
```

**Comment by jaapmarcus** [comment]
It looks like launchpad was down for the moment a while ago ... 

Please try again..

**Comment by leandroboari** [comment]
Sorry, but the problem still persists, but a little different now. Could you please help me?

Adding required repositories to proceed with installation:

[ * ] NGINX
[ * ] PHP
[ * ] Apache2
[ * ] MariaDB
[ * ] Hestia Control Panel

Updating currently installed packages, please wait... W: GPG error: http://ppa.launchpad.net/ondrej/apache2/ubuntu jammy InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 4F4EA0AAE5267A6C
E: The repository 'http://ppa.launchpad.net/ondrej/apache2/ubuntu jammy InRelease' is not signed.
\
The installer is now downloading and installing all required packages.
NOTE: This process may take 10 to 15 minutes to complete, please wait...
E: Package 'php8.0-common' has no installation candidate
E: Unable to locate package php8.0-curl
E: Couldn't find any package by glob 'php8.0-curl'
E: Couldn't find any package by regex 'php8.0-curl'
E: Unable to locate package php8.0-mysql
E: Couldn't find any package by glob 'php8.0-mysql'
E: Couldn't find any package by regex 'php8.0-mysql'
E: Unable to locate package php8.0-imap
E: Couldn't find any package by glob 'php8.0-imap'
E: Couldn't find any package by regex 'php8.0-imap'
E: Unable to locate package php8.0-ldap
E: Couldn't find any package by glob 'php8.0-ldap'
E: Couldn't find any package by regex 'php8.0-ldap'
E: Unable to locate package php8.0-apcu
E: Couldn't find any package by glob 'php8.0-apcu'
E: Couldn't find any package by regex 'php8.0-apcu'
E: Unable to locate package php8.0-zip
E: Couldn't find any package by glob 'php8.0-zip'
E: Couldn't find any package by regex 'php8.0-zip'
E: Unable to locate package php8.0-bz2
E: Couldn't find any package by glob 'php8.0-bz2'
E: Couldn't find any package by regex 'php8.0-bz2'
E: Unable to locate package php8.0-cli
E: Couldn't find any package by glob 'php8.0-cli'
E: Couldn't find any package by regex 'php8.0-cli'
E: Unable to locate package php8.0-gd
E: Couldn't find any package by glob 'php8.0-gd'
E: Couldn't find any package by regex 'php8.0-gd'
E: Unable to locate package php8.0-imagick
E: Couldn't find any package by glob 'php8.0-imagick'
E: Couldn't find any package by regex 'php8.0-imagick'
E: Unable to locate package php8.0-intl
E: Couldn't find any package by glob 'php8.0-intl'
E: Couldn't find any package by regex 'php8.0-intl'
E: Unable to locate package php8.0-mbstring
E: Couldn't find any package by glob 'php8.0-mbstring'
E: Couldn't find any package by regex 'php8.0-mbstring'
E: Unable to locate package php8.0-opcache
E: Couldn't find any package by glob 'php8.0-opcache'
E: Couldn't find any package by regex 'php8.0-opcache'
E: Unable to locate package php8.0-pspell
E: Couldn't find any package by glob 'php8.0-pspell'
E: Couldn't find any package by regex 'php8.0-pspell'
E: Unable to locate package php8.0-readline
E: Couldn't find any package by glob 'php8.0-readline'
E: Couldn't find any package by regex 'php8.0-readline'
E: Unable to locate package php8.0-xml
E: Couldn't find any package by glob 'php8.0-xml'
E: Couldn't find any package by regex 'php8.0-xml'
E: Package 'php8.0-common' has no installation candidate
E: Unable to locate package php8.0-bcmath
E: Couldn't find any package by glob 'php8.0-bcmath'
E: Couldn't find any package by regex 'php8.0-bcmath'
E: Unable to locate package php8.0-cli
E: Couldn't find any package by glob 'php8.0-cli'
E: Couldn't find any package by regex 'php8.0-cli'
E: Unable to locate package php8.0-curl
E: Couldn't find any package by glob 'php8.0-curl'
E: Couldn't find any package by regex 'php8.0-curl'
E: Unable to locate package php8.0-fpm
E: Couldn't find any package by glob 'php8.0-fpm'
E: Couldn't find any package by regex 'php8.0-fpm'
E: Unable to locate package php8.0-gd
E: Couldn't find any package by glob 'php8.0-gd'
E: Couldn't find any package by regex 'php8.0-gd'
E: Unable to locate package php8.0-intl
E: Couldn't find any package by glob 'php8.0-intl'
E: Couldn't find any package by regex 'php8.0-intl'
E: Unable to locate package php8.0-mysql
E: Couldn't find any package by glob 'php8.0-mysql'
E: Couldn't find any package by regex 'php8.0-mysql'
E: Unable to locate package php8.0-soap
E: Couldn't find any package by glob 'php8.0-soap'
E: Couldn't find any package by regex 'php8.0-soap'
E: Unable to locate package php8.0-xml
E: Couldn't find any package by glob 'php8.0-xml'
E: Couldn't find any package by regex 'php8.0-xml'
E: Unable to locate package php8.0-zip
E: Couldn't find any package by glob 'php8.0-zip'
E: Couldn't find any package by regex 'php8.0-zip'
E: Unable to locate package php8.0-mbstring
E: Couldn't find any package by glob 'php8.0-mbstring'
E: Couldn't find any package by regex 'php8.0-mbstring'
E: Unable to locate package php8.0-bz2
E: Couldn't find any package by glob 'php8.0-bz2'
E: Couldn't find any package by regex 'php8.0-bz2'
E: Unable to locate package php8.0-pspell
E: Couldn't find any package by glob 'php8.0-pspell'
E: Couldn't find any package by regex 'php8.0-pspell'
E: Unable to locate package php8.0-imagick
E: Couldn't find any package by glob 'php8.0-imagick'
E: Couldn't find any package by regex 'php8.0-imagick'


========================================================================

[ * ] Configuring system settings...
[ * ] Configuring Hestia Control Panel...
cp: cannot stat '/usr/local/hestia/install/deb/sudo/admin': No such file or directory
chmod: cannot access '/etc/sudoers.d/admin': No such file or directory
cp: cannot stat '/usr/local/hestia/install/deb/logrotate/hestia': No such file or directory
ln: failed to create symbolic link '/usr/local/hestia/log': No such file or directory
touch: cannot touch '/usr/local/hestia/log/system.log': No such file or directory
touch: cannot touch '/usr/local/hestia/log/nginx-error.log': No such file or directory
touch: cannot touch '/usr/local/hestia/log/auth.log': No such file or directory
touch: cannot touch '/usr/local/hestia/log/backup.log': No such file or directory
chmod: cannot access '/usr/local/hestia/log': No such file or directory
chmod: cannot access '/var/log/hestia/*': No such file or directory
cp: cannot stat '/usr/local/hestia/install/deb/packages': No such file or directory
sed: can't read /usr/local/hestia/data/packages/*.pkg: No such file or directory
cp: cannot stat '/usr/local/hestia/install/deb/templates': No such file or directory
cp: cannot stat '/usr/local/hestia/install/deb/templates/web/unassigned/index.html': No such file or directory
cp: cannot stat '/usr/local/hestia/install/deb/templates/web/skel/document_errors/*': No such file or directory
cp: cannot stat '/usr/local/hestia/install/deb/firewall': No such file or directory
cp: cannot stat '/usr/local/hestia/install/deb/api': No such file or directory
[ * ] Generating default self-signed SSL certificate...
hst-install-ubuntu.sh: line 1301: /usr/local/hestia/bin/v-generate-ssl-cert: No such file or directory
[ * ] Adding SSL certificate to Hestia Control Panel...
sed: -e expression #1, char 3: unexpected `,'
sed: -e expression #1, char 1: unknown command: `,'
cp: cannot stat '/usr/local/hestia/install/deb/ssl/dhparam.pem': No such file or directory
[ * ] Enable SFTP jail...
Error: can't enable sftp jail

**Comment by jaapmarcus** [comment]
Just tried an install on Ubuntu 22.04

```Would you like to continue with the installation? [Y/N]: y

Please use a valid emailadress (ex. info@domain.tld).
Please enter admin email address: me@eris.nu
Please enter FQDN hostname [demo.eris.nu]: 
Installation backup directory: /root/hst_install_backups/281120222104
Installation log file: /root/hst_install_backups/hst_install-281120222104.log

Adding required repositories to proceed with installation:

[ * ] NGINX
[ * ] PHP
[ * ] Apache2
[ * ] MariaDB
[ * ] Hestia Control Panel

Updating currently installed packages, please wait...\
The installer is now downloading and installing all required packages.
NOTE: This process may take 10 to 15 minutes to complete, please wait... 
Extracting templates from packages: 100%


========================================================================

[ * ] Configuring system settings...
[ * ] Configuring Hestia Control Panel...
[ * ] Generating default self-signed SSL certificate...
[ * ] Adding SSL certificate to Hestia Control Panel...
[ * ] Enable SFTP jail...
[ * ] Create admin account...
[ * ] Configuring NGINX...
[ * ] Configuring Apache Web Server...
[ * ] Install  PHP 8.0...
[ * ] Configuring PHP-FPM 8.0...

```

**Comment by jaapmarcus** [comment]
Maybe Create an issue:
https://github.com/oerdnj/deb.sury.org/issues

He is the maintainer of the repo. :heart:

**Comment by leandroboari** [comment]
Could you please check in version 22.04.1? The problem appears in the installer... Other users also face the same problem.

https://forum.hestiacp.com/t/enable-sftp-jail-error-cant-enable-sftp-jail/7225

Distributor ID: Ubuntu
Description:    Ubuntu 22.04.1 LTS
Release:        22.04
Codename:       jammy

**Comment by jaapmarcus** [comment]
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.1 LTS
Release:	22.04
Codename:	jammy


Again the issue is caused by the fact that  PHP repo has not been installed properly and there for not all packages are installed. Including the Hestia packages... 

The error the other user had is not related to this. 

```
Updating currently installed packages, please wait... W: GPG error: http://ppa.launchpad.net/ondrej/apache2/ubuntu jammy InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 4F4EA0AAE5267A6C
```

This is the main issue. Maybe ppa launchpad or the key server was offline for a moment.  But this is out of control for Hestia it self. This Is an issue with Launchpad / ppa. 

You should ask the maintainer of PHP packages and solve the issue there..
